agent/internal/client/methods.go

```diff
@@ -579,6 +579,23 @@ func (c *communicatorImpl) SendTestResults(ctx context.Context, taskData TaskDat
 	return nil
 }
 
+// SetHasResults sets the HasResults flag to true in the given task in the
+// database.
+func (c *communicatorImpl) SetHasResults(ctx context.Context, taskData TaskData) error {
+	info := requestInfo{
+		method:   http.MethodPost,
+		taskData: &taskData,
+		version:  apiVersion1,
```

Comment:
Use API version 2.

Reply:
Would this require moving the route to the `rest` package? I feel like it makes to leave in `service` cause that's where all the agent commands are.

Reply:
At this point, new routes should be written in `rest/route`. The ones still in the legacy API can remain there but we shouldn't add more functionality to it.